### DISCUSSION TITLE
Filter on Release.Edited but NOT While Published

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [community/community#25183] [Actions]
<p>Hello All,</p>
<p>I have the following workflow file:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/blob/master/.github/workflows/update-changelog.yml" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/blob/master/.github/workflows/update-changelog.yml</a></p>
<p>Everything works great except for when I publish a draft release, the <code>edited</code> event seems to fire as well.  Is there a way to only trigger this workflow when the current type is NOT <code>published</code>?</p>
<p>Thank you for any assistance you can provide. :+1:</p>

**Comment by BrightRan** [comment, marked as answer]
<p>I tested with the same event trigger configuration as yours, and then tried the following actions:</p>
<ol>
<li>Drafted a new release, edited and published directly, the workflow was not triggered as expected.</li>
</ol>
<p>2. Drafted a new release, edited and published as a pre-release, the workflow was not triggered as expected.</p>
<p>3. Drafted a new release, edited and save draft, re-edited the draft and re-save the draft, then published draft, the workflow was not triggered as expected.</p>
<ol start="4">
<li>
<p>Drafted a new release, edited and save draft, re-edited the draft and published draft, the workflow was not triggered as expected.</p>
</li>
<li>
<p>Opened the published release in <strong>action 1</strong> , re-edited the published release and then click “Update release”, the workflow was triggered as expected.</p>
</li>
</ol>
<p>6. Opened the published pre-release in <strong>action 2</strong> , re-edited the published pre-release and then click “Update release”, the workflow was triggered as expected.</p>
<p>According to my above tests, the workflow only can be triggered when the published releases or pre-releases are updated/edited, this looks like is normal without problems.  <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#release-event-release" rel="nofollow noopener">Note: The release event is not triggered for draft releases.</a></p>
<p>If you want to know what triggers your workflow run, you can use the below two properties of <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">github context</a> to get more info:</p>
<p><strong>github.event_name</strong> – the name of the event, for example:  <strong>release</strong>.</p>
<p><strong>github.event.action</strong> – the action type of the event, for example:  <strong>edited</strong>.</p>
<p>And you aslo can reference the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#example-printing-context-information-to-the-log-file" rel="nofollow noopener">example printing context information to the log file</a> to print the detailed info of github context.</p>
<p><img src="https://user-images.githubusercontent.com/46307996/181089280-62edf1ed-faaf-4711-bedf-ab5f1ba978ff.png" alt="example.png" data-base62-sha1="1qzvrrHFjbFbqqCrLH1az04BFgv" width="395" height="303"></p>

**Comment by Mike-E-angelo** [comment]
<p>Thank you for your reply <a class="mention" href="/brightran">@brightran</a>!</p>
<p>Is there a reason why this information is not printed out by default?  It seems *very* important!</p>
<p>FWIW the run in question is here: <a href="https://github.com/ExtendedXmlSerializer/home/actions/runs/35826091" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/actions/runs/35826091</a></p>
<p>I am not sure if you are able to see with your own records what caused that, but it would be nice to know what happened if possible.</p>
<p>In any case, I will add the output to my workflow, thanks again.  Hopefully, we can see going forward what has caused the workflow to trigger and can further debug from there.</p>

**Comment by Mike-E-angelo** [comment]
<p>Also <a class="mention" href="/brightran">@brightran</a>, when you share YML as part of a solution, can you please share it as text and not an image?  It makes it a bit easier to copy/paste into our solutions. :grin:</p>

**Comment by BrightRan** [comment]
<p>All right, sorry for that.</p>
<p>How are things going now?</p>
<p>Have you tried the suggestions I mentioned above? Are they helpful for you to solve the problem?</p>

**Comment by Mike-E-angelo** [comment]
<p>Ah I won’t know until our next release, unfortunately, but have put the YML in here in case anyone else wants to copy/paste it:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/blob/master/.github/workflows/publish-release-build.yml#L17-L25" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/blob/master/.github/workflows/publish-release-build.yml#L17-L25</a></p>
<p>I will certainly update here on our next release and let you know of any further issues we run into… thank you for your assistance, <a class="mention" href="/brightran">@brightran</a> ! :+1:</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mike-e-wins">@mike-e-wins</a> ,</p>
<p>Have you tried as my suggestion that using the properties <strong>github.event_name</strong> and <strong>github.event.action</strong> of github context to get more info about the event?<br>
<strong>github.event_name</strong> – the name of the event, for example: <strong>release</strong>.<br>
<strong>github.event.action</strong> – the action type of the event, for example: <strong>edited</strong>.</p>
<p>How are things going? Is my suggestion helpful to you?<br>
Any progress, please feel free to tell me.</p>

**Comment by Mike-E-angelo** [comment]
<p>Hey <a class="mention" href="/brightran">@brightran</a> !  As luck would have it, we have not released a new release for some time now.  So, I am not sure. :laughing:  I will certainly update here and let you know.  You can watch our releases here and know when a new release is available:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/releases" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/releases</a></p>
<p>But I will be sure to let you know if I release and/or any troubles we find in doing so. :+1:</p>

**Comment by Mike-E-angelo** [comment]
<p>Alright <a class="mention" href="/brightran">@brightran</a> wanted to give you a heads up we are about to launch another version here next week.  Looking at Tuesday.  We will be able to further proceed then.  Thank you for your patience and again for your assistance!</p>

**Comment by Mike-E-angelo** [comment]
<p>Alright <a class="mention" href="/brightran">@brightran</a> I created a new release and the good news is that it worked!  Looks like the <code>published</code> event is now being emitted:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/runs/533423690?check_suite_focus=true" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/runs/533423690?check_suite_focus=true</a></p>
<p>Unfortunately, I have a new problem that hasn’t been there before:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/runs/533424140?check_suite_focus=true" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/runs/533424140?check_suite_focus=true</a></p>
<pre><code class="lang-auto">
Installing package 'Poshstache' Downloaded 0.00 MB out of 0.29 MB. [] ::error file=/github/home/.local/share/powershell/Modules/Poshstache/0.1.6/ConvertTo-PoshstacheTemplate.ps1,line=67,col=24::Unable to find type [Stubble.Core.Builders.StubbleBuilder].

</code></pre>
<p>Do you happen to know anything about this?</p>

**Comment by Mike-E-angelo** [comment]
<p>FWIW I have asked the above to the general community here:</p>
<p><a href="https://github.community/t/new-exception-on-running-workflow/17947">https://github.community/t5/GitHub-Actions/New-Exception-on-Running-Workflow/m-p/51304#M8142</a></p>
